core/src/main/scala/org/apache/spark/util/SparkConfigParseUtils.scala

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.util
+
+import java.io.PrintStream
+
+/*
+ * Provides logic to parse Spark config properties from cli.
+ * Used by processes creating a SparkConfig from cli args.
+ * Facilitates the use of common error messages when spark conf parsing fails.
+ *
+ */
+private[spark] trait SparkConfigParseUtils {
```

Comment:
I'm not really in love with the name of this class. Not only it's awkward, but it doesn't reflect the actual functionality, since it has more than just utility methods for parsing configs from the command line.

I'd call it something more generic like `CommandLineUtils`, saying that it contains basic command line parsing functionality and methods to parse some common Spark CLI options.